src/Jackalope/Transport/DoctrineDBAL/Client.php

```
@@ -1679,10 +1680,19 @@ private function moveNode(string $srcAbsPath, string $destAbsPath): void
             throw new PathNotFoundException("Parent of the destination path '".$destAbsPath."' has to exist.");
         }
 
-        $query = 'SELECT path, id FROM phpcr_nodes WHERE path LIKE ? OR path = ? AND workspace_name = ? '.$this->getConnection()
-                ->getDatabasePlatform()->getForUpdateSQL()
+        $sql = $this->getConnection()->createQueryBuilder()
+            ->select('path, id')
+            ->from('phpcr_nodes')
+            ->where('(path LIKE :path_prefix OR path = :path) AND workspace_name = :workspace')
+            ->forUpdate()
```

@derrabus did i understand your suggestion in https://github.com/jackalope/jackalope-doctrine-dbal/pull/436#discussion_r1470991804 wrong?

this breaks sqlite with "FOR UPDATE" not supported errors. or is that a bug in dbal and it is supposed to correctly decide if it should put FOR UPDATE into the statement?

---

The purpose of that method is to pick the right syntax for the lock hint. However, if your database does not support this kind of locks, the method does not silently fall back to no locking at all.

The idea is that it should be a deliberate decision on your side to not lock.

```php
        $qb = $this->getConnection()->createQueryBuilder()
            ->select('path, id')
            ->from('phpcr_nodes')
            ->where('(path LIKE :path_prefix OR path = :path) AND workspace_name = :workspace')
        ;

        if (!$this->getConnection()->getDatabasePlatform instanceof SQLitePlatform) {
            $qb->forUpdate();
        }

        $sql = $qb->getSQL();
```

But you're not the first to run into this problem. Maybe we will implement a configurable fallback strategy on our side.

---

this still happens for sqlite, now that reverting mysql version and encoding got us rid of the fatal errors.

---

okay, i see. in the context of the discussion (where we just got it from whichever platform was being used and got either nothing or the correct statement) i thought the query builder would be aware of the platform.

the unfortunate thing is that this makes the application think about specific implementations and have assumptions. it would fail with custom platform implementations or new implementations that also do not support the lock hint, or if some future sqlite version would support it...

thanks for the explanation, then i will hardcode the check for now.

---

> i thought the query builder would be aware of the platform.

It is and this is why you get an exception. 🙃 

> the unfortunate thing is that this makes the application think about specific implementations and have assumptions. it would fail with custom platform implementations or new implementations that also do not support the lock hint, or if some future sqlite version would support it...

Yes, it would. I understand that this is a problem for you, hence the last paragraph of my previous comment. 🙂